### PR TITLE
Create scaffold

### PR DESCRIPTION
**Pull request by jinkang-0**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Created scaffold for sign up and applications
- Created custom `Link` component for styling, expandable for other variants in the future
- Added missing packages for ESLint so it doesn't error
- Modified project spec to use LF (so Prettier doesn't complain)

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

Home:
image

Applications:
image

Application Detail:
image

Sign up page
image

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Test the pages work as expected and that you can navigate to them. Routes should be `/`, `/applications`, `/applications/[appId]`, and `/sign-up`.

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

Fixing ESLint: https://github.com/vercel/next.js/issues/73968
Dynamic parameter parsing: https://nextjs.org/docs/app/api-reference/file-conventions/dynamic-routes#convention

CC: @ethan-tam33

### CODE REVIEW

File: hooks/application.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    const stages = ['pre', 'main', 'matches', 'submitted'] as const;
+    const currentStage = searchParams.get('stage') || 'pre';
+    const highestAttainedStage = appState.highestStageAchieved;
+
+    // parse current and highest achieved stages as index
+    const currentIdx = stages.findIndex(stage => stage === currentStage);
+    const highestIdx = stages.findIndex(
+      stage => stage === highestAttainedStage,
+    );
+    if (currentIdx === -1 || highestIdx === -1) return;
```
[the remark this answers is not in the document]
Comment (jinkang-0) [reply hooks/application.tsx]: yup, fixed the "go back" button and added some descriptive errors.

**Review by ethan-tam33 — APPROVED** [review]
